src/main/java/com/iopipe/generic/EntryPoint.java

```diff
@@ -577,5 +505,271 @@ else if (pn == 0)
+		 * @param __t The type to resolve.
+		 * @throws NullPointerException On null arguments.
+		 * @since 2018/09/05
+		 */
+		static Type __resolve(Deque<Class<?>> __cls, Type __t)
+			throws NullPointerException
+		{
+			if (__cls == null || __t == null)
+				throw new NullPointerException();
+			
+			// If it is not a type variable, just return self
+			if (!(__t instanceof TypeVariable))
+				return __t;
+			
+			// Make a copy of the stack
+			Deque<Class<?>> stackcopy = new ArrayDeque<>(__cls);
+			
+			TypeVariable tv = (TypeVariable)__t;
+			String tvname = tv.getName();
+			
+			// The class that declares this type parameter was the
+			// one the method was found in (it will be at the top)
+			Class<?> pivotclass = stackcopy.pop();
+			
+			// Determine which index our type variable is in, we
+			// need to look in the super class
+			TypeVariable<?>[] pivotvars = pivotclass.getTypeParameters();
+			int pivotdx = -1;
+			for (int j = 0, n = pivotvars.length; j < n; j++)
+				if (tvname.equals(pivotvars[j].getName()))
+				{
+					pivotdx = j;
+					break;
+				}
+			
+			// This should not happen normally (unless the class
+			// was modified to break it or we are trying to initialize
+			// a non-static inner class which cannot be initialized
+			// anyway)
+			if (pivotdx < 0)
+				return __t;
+			
+			// Peek the current class
+			Class<?> upperclass = stackcopy.peek();
+			
+			// Initializing a class which is just a type variable
+			if (upperclass == null)
+				return __t;
+			
+			// Need to go through all types and determine what this
+			// is
+			Type[] genints = upperclass.getGenericInterfaces();
+			for (int j = 0, n = genints.length; j <= n; j++)
+			{
+				Type maybe = (j == 0 ?
+					upperclass.getGenericSuperclass() :
+					genints[j - 1]);
+				
+				if (!(maybe instanceof ParameterizedType))
+					continue;
+				
+				// Only look at an extend which is of our own class
+				ParameterizedType pt = (ParameterizedType)maybe;
+				if (!pivotclass.equals(ObjectTranslator.__rawClass(pt)))
+					continue;
+				
+				// This could fail
+				try
+				{
+					Type[] ptparms = pt.getActualTypeArguments();
+					int nptparms = ptparms.length;
+					
+					// This should not happen normally but if it does it means
+					// the parent class had a change to its type parameters
+					// which did not cause a compilation failure
+					if (j >= nptparms)
+						break;
+					
+					// Recursively resolve that parameter because it could just
+					// be another type parameter
+					// Remember to use our copy of the stack since we removed
+					// some elements from it
+					return __Target__.__resolve(stackcopy, ptparms[j]);
+				}
+				
+				// There is something wrong with the parameter, ignore
+				catch (TypeNotPresentException|MalformedParameterizedTypeException e)
+				{
+				}
+				
+				// We found our class so we do not need to look elsewhere
+				// however something was wrong and we could not determine
```

Comment:
Reword to say that "We found our class, and we found no type that could be used. All other checks will fail so we just stop.".